Keen.NetStandard/EventCachePortable.cs

```diff
@@ -0,0 +1,201 @@
+﻿using Keen.NetStandard.EventCache;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using System.IO;
+using System.Text;
+
+namespace Keen.NetStandard
+{
+    /// <summary>
+    /// <para>EventCachePortable implements the IEventCache interface using
+    /// file-based storage. It has no cache-expiration policy.</para>
+    /// <para>To use, pass an instance of this class when constructing KeenClient.
+    /// To construct a new instance, call the static New() method.</para>
+    /// </summary>
+    public class EventCachePortable : IEventCache
+    {
+        // An asynchronous, lazy loaded instance of this class.
+        private static AsyncLazy<EventCachePortable> _instanceTask =
+            new AsyncLazy<EventCachePortable>(
+                taskFactory: CreateInstanceAsync);
+
+        // A list of events in the cache
+        protected List<string> events = new List<string>();
+
+        protected EventCachePortable() { }
+
+        /// <summary>
+        /// Get the singleton instance of EventCachePortable.
+        /// Since initialization is async, this wraps the instance
+        /// with a Task.
+        /// </summary>
+        /// <para>To get the EventCachePortable instance, await this property.</para>
+        /// <para>The Task<EventCachePortable> is available via Value on this property.</para>
+        /// <returns></returns>
+        public static AsyncLazy<EventCachePortable> InstanceTask => _instanceTask;
```

Comment:
Then again if it's awaitable since it has a `GetAwaiter()`, is the idea that client code just calls `await EventCachePortable.InstanceTask` because in that case it's all groovy 👍